sklearn/preprocessing/_encoders.py

```diff
@@ -41,6 +41,9 @@ def _check_X(self, X):
           not do that)
 
         """
+        if hasattr(X, 'iloc'):
```

Comment:
If we're going to start supporting pandas in a few places, we should introduce a `is_pandas_df()` or something in `sklearn.utils.validation` maybe, I guess.

Reply:
Note `hasattr(X, 'iloc')` would be true of a Series too

Reply:
Yes, sorry, here is the link to the other (pandas related) PR: https://github.com/scikit-learn/scikit-learn/pull/13253